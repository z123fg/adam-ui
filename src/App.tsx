import React from "react";
import "./styles/index.scss";
import MyButton, {
  ButtonSize,
  ButtonType,
} from "./components/MyButton/MyButton";
import MyModal from "./components/MyModal/MyModal";

function App() {
  return (
    <div className="App">
      <h1>Adam UI</h1>
      <h4>Default</h4>
      <MyButton>Default</MyButton>
      <MyButton disabled>Disabled</MyButton>
      <MyButton isDark>Dark Button</MyButton>
      <MyButton className="btn-red">Additional Class Name</MyButton>
      <h4>Sizes</h4>
      <MyButton buttonSize={ButtonSize.Small}>Small Button</MyButton>
      <MyButton buttonSize={ButtonSize.Large}>Large Button</MyButton>
      <h4>Types</h4>
      <MyButton >Default Button</MyButton>
      <MyButton buttonType={ButtonType.Primary}>Primary Button</MyButton>
      <MyButton buttonType={ButtonType.Secondary}>Secondary Button</MyButton>
      <MyButton buttonType={ButtonType.Danger}>Danger Button</MyButton>
      <MyButton href={"https://www.google.com"}  buttonType={ButtonType.Link}>Link Button</MyButton>
      <MyButton  href={"https://www.google.com"} buttonType={ButtonType.Link} disabled>Link Button</MyButton>
      <MyModal />
    </div>
  );
}

export default App;
