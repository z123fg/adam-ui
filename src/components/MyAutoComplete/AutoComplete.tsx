import * as React from "react";

interface Props {
    data: string[];
    onSelect?: Function
}

const AutoComplete: React.FC<Props> = ({ data, onSelect }) => {
    const [search, setSearch] = React.useState<string>("");
    const [filterResults, setFilterResults] = React.useState<string[]>([]);
    const [showResults, setShowResults] = React.useState<boolean>(false);

    const filterSupport = (input:string) => {
        const result = data.filter((value) => 
        value.toLowerCase().indexOf(input.toLowerCase()) > -1);

        if(input.length > 0) {
            setFilterResults(result);
        } else {
            setFilterResults([]);
        }

    }

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const input = e.target.value;
        setSearch(input)
        filterSupport(input)
        setShowResults(true);
    };

    const handleSelect = (itemSelected:string) => {
        setShowResults(false);
        setFilterResults([]);
        setSearch(itemSelected)
        if(onSelect !== undefined){
            onSelect(itemSelected)
        }
    }

    const renderSuggestions = () => {
        return filterResults.length?(
            <>
                <ul className="ac-list">
                    {filterResults.map((result:string, index:number) => {
                        return <li key={index} onClick={()=>handleSelect(result)}>{result}</li>
                    })}
                </ul>
            </>
        ) : (
            <>
                <p className="ac-no-result">Sorry, no results found.</p>  
            </>   
        )
    }

    return (
        <>
            <input type="text" value={search} onChange={handleChange} className="ac-input"/>
            {showResults && search && renderSuggestions()}
        </>
    )
}

export default AutoComplete