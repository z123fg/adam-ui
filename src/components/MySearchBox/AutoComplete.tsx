import * as React from "react";

interface Props {
    data: string[];
}

const AutoComplete: React.FC<Props> = ({ data }) => {
    const [search, setSearch] = React.useState<string>("");
    const [filterResults, setFilterResults] = React.useState<string[]>([]);
    const [showResults, setShowResults] = React.useState<boolean>(false);

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const input = e.target.value;
        setSearch(input)

        const result = data.filter((value) => 
            value.toLowerCase().indexOf(input.toLowerCase()) > -1
        );

        if(input.length > 0) {
            setFilterResults(result);
        } else {
            setFilterResults([]);
        }

        setShowResults(true);
    };

    const handleSelect = (result:string) => {
        setShowResults(false);
        setFilterResults([]);
        setSearch(result)
    }

    const Suggestions = () => {
        return filterResults.length?(
            <>
                <ul className="ac-list">
                    {filterResults.map((result:string, index:number) => {
                        return <li key={index} onClick={()=>handleSelect(result)}>{result}</li>
                    })}
                </ul>
            </>
        ) : (
            <>
                <p className="ac-no-result">Sorry, no results found.</p>  
            </>   
        )
    }

    return (
        <>
            <input type="text" value={search} onChange={handleChange} className="ac-input"/>
            {showResults && search && <Suggestions />}
        </>
    )
}

export default AutoComplete