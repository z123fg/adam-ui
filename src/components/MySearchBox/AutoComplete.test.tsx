import React from "react";
import ReactDOM from 'react-dom';
import AutoComplete from "./AutoComplete";
import App from "../../App"
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from "@testing-library/react";

describe("test MyButton Component", () => {
  // check if the app component renders without crashing
  it('renders without crashing', () => {
    render
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // check if the ui renders as expected
  it('renders the UI as expected', () => {
    const data = ["1", "2", "3"];
    const tree = renderer
      .create(<AutoComplete data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
