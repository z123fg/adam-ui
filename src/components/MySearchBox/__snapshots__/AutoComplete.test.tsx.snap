// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyButton Component renders the UI as expected 1`] = `
<input
  className="ac-input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
